README.md

```diff
@@ -1,16 +1,140 @@
+- **Server-side Rendering (SSR):** Next.js is used to provide server-side rendering, improving initial load times and enhancing the overall performance of the application.
+- **Component-based Architecture:** The application follows a component-based architecture, promoting code reusability and modular development.
+- **State Management:** The application utilizes React's built-in state management capabilities, along with Redux for managing global application state when necessary.
+- **Testing:** The codebase includes unit tests and integration tests, ensuring code quality and facilitating easier maintenance and refactoring.
+- **Modular Architecture:** The codebase is structured using a modular approach, promoting code reusability, maintainability, and scalability.
+
+![High-Level-Diagram](image.png)
+
+## Getting Started
+
+### Prerequisites
+
+- [Node.js] (https://nodejs.org/en)
+- [Firebase] (https://firebase.google.com/): Account for their Authentication and Picture Storage
+- [Protomaps] (https://protomaps.com/dashboard): API Key
+
+### Installation
+
+1. Fork the repository on GitHub.
+2. Clone your forked repository:
+3. Navigate to the project directory:
+4. Install dependencies:
+
+```
+npm install
+```
+
+### Firebase - Setup
+
+1. Install Firebase with npm install -g firebase-tools. To check your current version of Firebase, run the command firebase --version.
+
+2. The installation is done globally, so if you encounter an firebase: command not found error, be sure to update your PATH variable on your machine's environmental variables or run npm exec -- firebase [CLI args].
+
+3. Log into Firebase with firebase login using the account that has the Firebase project.
+   3.1 Initialize Firebase with firebase init, and set up the Authentication emulator.
+   3.2 If a .firebaserc file already exists, run the command firebase init emulators instead.
+
+4. Set your firebaseConfig object variables in .env. These values can be found in the Google Firebase Console: Project Settings → Your app → SDK setup and configuration.
+5. Run the command firebase emulators:start.
+
+### Development
+
+To start the development server, run:
+
+```
+npm run dev
+```
+
+The application will be available at `https://localhost:3000`.
+
+### Setting up the Enviromental Variables
+
+Go to .env.local.example folder and replace the placeholders with your keys
+
+### Building for Production
+
+To build the application for production, run:
+
+```
+npm run build
+```
+
+After the build is complete, you can start the production server with:
+
+```
+npm start
+```
+
+### Linting and Testing
+
+To lint the codebase, run:
+
+```
+npm run lint
```

Comment:
Do we have the ability to lint our codebase? This I didn't know.

Reply:
we can run the linter with npx next lint. Changed it in the read me to that.